android/app/src/main/java/com/zulipmobile/sharing/ReceiveShareActivity.kt

```diff
@@ -0,0 +1,129 @@
+import android.os.Bundle
+import android.os.Parcelable
+import android.util.Log
+import android.webkit.WebView
+import androidx.annotation.Nullable
+import com.facebook.react.ReactActivity
+import com.facebook.react.ReactApplication
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.Arguments
+import com.zulipmobile.MainActivity
+import com.zulipmobile.notifications.*
+
+const val TAG = "ZulipReceiveShare"
+
+class ReceiveShareActivity : ReactActivity() {
+
+    /**
+     * Returns the name of the main component registered from JavaScript.
+     * This is used to schedule rendering of the component.
+     */
+    override fun getMainComponentName(): String? {
+        return "SharingRoot"
+    }
+
+    private fun sendEvent(reactContext: ReactContext,
+                          eventName: String,
+                          @Nullable params: WritableMap) {
+        Log.d(TAG, "Sending event with shared data")
+        emit(reactContext, eventName, params)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        WebView.setWebContentsDebuggingEnabled(true)
+        val application = application as ReactApplication
+        launchMainActivity(application as Context)
+        val host = application.reactNativeHost
+        val reactContext = host.tryGetReactInstanceManager()?.currentReactContext
+        val intent = intent
+        when {
+            intent?.action == Intent.ACTION_SEND -> {
+                when {
+                    "text/plain" == intent.type -> handleSendText(reactContext, intent)
+                    intent.type?.startsWith("image/") == true -> handleSendImage(reactContext, intent)
+                    else -> handleGenericFile(reactContext, intent)
+                }
+            }
+        }
+        finish()
+    }
+
+    private fun handleSendText(reactContext: ReactContext?, intent: Intent) {
+        val sharedText = intent.getStringExtra(Intent.EXTRA_TEXT)
+        val params = Arguments.createMap()
+        params.putString("type", "text")
+        params.putString("sharedText", sharedText)
+        if (sharedText != null) {
+            if (reactContext == null) {
+                SharingModule.initialSharedData = params
+            } else {
+                sendEvent(reactContext, "shareReceived", params)
+            }
+
+        }
+    }
+
+    private fun handleSendImage(reactContext: ReactContext?, intent: Intent) {
+        (intent.getParcelableExtra<Parcelable>(Intent.EXTRA_STREAM) as? Uri)?.let {
+            val params = Arguments.createMap()
+            params.putString("type", "image")
+            params.putString("sharedImageUri", it.toString())
+            if (reactContext == null) {
+                SharingModule.initialSharedData = params
+            } else {
+                sendEvent(reactContext, "shareReceived", params)
+            }
+        }
```

Comment:
No -- that wouldn't get uniformity, because more things use "URL" anyway.

Quick empirical demo:
```
$ grep -Rhi url node_modules/react-native | wc -l
2142

$ grep -Rhi uri node_modules/react-native | wc -l
829
```

A lot of people chose "URI" as the name for various things because it was officially the standard for a number of years. But it was never the name most people would normally use in a sentence -- most people said "URL", and would say "URI" only if they were really trying hard to be pedantically standards-correct. On top of that, the official line for all those years was that "URL" totally *was* the name of a subset of URIs that just happened to encompass virtually all of the URIs that anyone ever encountered in practice.

So now that even the relevant current standard recommends dropping "URI", it really has nothing going for it at all outside of contexts with specific legacy constraints.

In some places we do have those legacy constraints: so for example in the Zulip server API, which is hard to make changes to because there are lots of existing servers and clients, there's a field called `realm_uri`. Internally we have things we call `realmUri` because they correspond very directly to a value of that field.

But we shouldn't make *new* API choices that propagate the old confusing name. "URL" is the winner. Fighting it just prolongs the confusion.

Concretely, this key `sharedImageUri` in a data blob we're sending here over to JS is a new API choice, in the internal API between the Kotlin and JS sides of our app. So we should use the standard name for that new choice.

For a local variable, it doesn't matter as much. But the only way to get toward uniformity is to make a habit of picking the standard name, when there isn't a strong reason to go the other way.

Reply:
Changed relevant names to `uri`.